### PR TITLE
RHEL 8.5: tar installer

### PR DESCRIPTION
**Pull request by achilleas-k**
This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

Implements a `tar-installer`: an installer image that embeds an OS tarball (same as the `tar` image type) that's installed when the ISO is booted.

I had this implemented and ready for a while based on #1351.  Rebased now on #1391.  Opening now for early review but keeping as a draft until we #1391 is merged and to add tests.

### CODE REVIEW

File: internal/distro/rhel85/distro_test.go
Comment (atodorov) [inline internal/distro/rhel85/distro_test.go:37]: From what I can see this is verifying that we've got these 2 new image types. However I don't see any test which tries to build the new image types and makes sure nothing fails. `cmd/osbuild-image-tests/main_test.go` may be a good place to start. Alternatively a separate integration test would do as well.

Comment (achilleas-k) [reply internal/distro/rhel85/distro_test.go:37]: Thanks for the review, Alex.  I was thinking an integration test that builds the installer and auto-installs it, similar to the ostree integration tests, would be a good idea for the tar-installer.

The plain tar image type existed in the previous distros, so we could follow the same testing here.

**Comment by achilleas-k** [comment]
Rebased on `main` and added an integration test that builds an installer, installs it and checks for installed packages (I based it on `ostree.sh`).

**Comment by achilleas-k** [comment]
@atodorov or @ondrejbudai I haven't enabled the integration test in the `.gitlab-ci.yml` file yet.  Is the 8.5 runner ready?

**Comment by teg** [comment]
Really nice @achilleas-k !

@jikortus is this something you could use/extend as part of your work?

**Review by atodorov — CHANGES_REQUESTED** [review]
@achilleas-k you should be able to rebase to latest main and see your test jobs running on the 8.5 runners.

File: .gitlab-ci.yml
Comment (atodorov) [inline .gitlab-ci.yml:236]: This has been removed by @ondrejbudai in #1510. You can drop the EXTRA_REPO_PATH_SEGMENT variable.

File: .gitlab-ci.yml
Comment (atodorov) [inline .gitlab-ci.yml:84]: This stage is used when testing the latest nightly builds so IMO use only 8.5, not both.

**Review by atodorov — CHANGES_REQUESTED** [review]
Minor changes requested, otherwise looks good.

File: .gitlab-ci.yml
Comment (achilleas-k) [reply .gitlab-ci.yml]: Removed.  Thanks.

File: .gitlab-ci.yml
Comment (achilleas-k) [reply .gitlab-ci.yml:83]: Switched to 8.5 only.

**Comment by achilleas-k** [comment]
Rebased and resolved conflicts.

**Comment by achilleas-k** [comment]
Had to update the test manifests for the edge image types introduced in #1351.  The core stages that are shared between all image types didn't include the fix-bls stage before the installer type was added.

File: internal/distro/rhel85/distro_test.go
Comment (atodorov) [reply internal/distro/rhel85/distro_test.go:232]: @achilleas-k please open a ticket in JIRA with the `QE` prefix for this tar integration test if you haven't already so that we don't forget about it and make sure to plan for the work accordingly.

File: .gitlab-ci.yml
Comment (atodorov) [inline .gitlab-ci.yml:110]: FTR this is going to fail b/c of the other missing images. Once the CI job passes we should vet the tests which failed, there should be only a few of them which either use qcow2 or try to compare the list of available image types. Everything else should report PASS.

File: internal/distro/rhel85/distro_test.go
Comment (atodorov) [reply internal/distro/rhel85/distro_test.go:232]: ^^^ I see installers.sh. Closing this comment.

**Review by atodorov — APPROVED** [review]
LGTM with comments.

File: .gitlab-ci.yml
Comment (achilleas-k) [reply .gitlab-ci.yml:110]: What's the plan then?  Are we keeping this open until all the other image types land?  I don't mind either way.

Comment (teg) [reply .gitlab-ci.yml:110]: I guess the plan is up to you ;) Would it be a lot of churn to only enable the tests we expect to pass for this PR, and then enable the rest when the other image types are ready?

Comment (atodorov) [reply .gitlab-ci.yml:110]: @achilleas-k let's not keep PRs open longer than they need to. The fact that there's missing support for image types (which has the nice side effect of breaking our test suite) is not related to the work around tar installer.

**Comment by ondrejbudai** [comment]
Please, don't merge PRs that are clearly untested by the CI. For this PR, none of the tests were actually run because the CI actually failed on the RPM build step. I hope this won't break `main`.

image

**Comment by ondrejbudai** [comment]
Just a note why it failed on rpms: This was caused by the outage of RHSM servers.

**Comment by atodorov** [comment]
My bad, sorry. I didn't inspect the GitLab results this time and relied on the last time I saw things were looking good.

**Pull request merged**
